src/main/java/io/lettuce/core/AbstractRedisAsyncCommands.java

```diff
@@ -1453,6 +1467,176 @@ public boolean isOpen() {
         return connection.isOpen();
     }
 
+    @Override
+    public RedisFuture<List<Long>> jsonArrappend(K key, JsonPath jsonPath, JsonValue... values) {
+        return dispatch(jsonCommandBuilder.jsonArrappend(key, jsonPath, values));
+    }
+
+    @Override
+    public RedisFuture<List<Long>> jsonArrappend(K key, JsonValue... values) {
+        return dispatch(jsonCommandBuilder.jsonArrappend(key, JsonPath.ROOT_PATH, values));
+    }
+
+    @Override
+    public RedisFuture<List<Long>> jsonArrindex(K key, JsonPath jsonPath, JsonValue value, JsonRangeArgs range) {
+        return dispatch(jsonCommandBuilder.jsonArrindex(key, jsonPath, value, range));
+    }
+
+    @Override
+    public RedisFuture<List<Long>> jsonArrindex(K key, JsonPath jsonPath, JsonValue value) {
+        return dispatch(jsonCommandBuilder.jsonArrindex(key, jsonPath, value, JsonRangeArgs.Builder.defaults()));
+    }
+
+    @Override
+    public RedisFuture<List<Long>> jsonArrinsert(K key, JsonPath jsonPath, int index, JsonValue... values) {
+        return dispatch(jsonCommandBuilder.jsonArrinsert(key, jsonPath, index, values));
+    }
+
+    @Override
+    public RedisFuture<List<Long>> jsonArrlen(K key, JsonPath jsonPath) {
+        return dispatch(jsonCommandBuilder.jsonArrlen(key, jsonPath));
+    }
+
+    @Override
+    public RedisFuture<List<Long>> jsonArrlen(K key) {
+        return dispatch(jsonCommandBuilder.jsonArrlen(key, JsonPath.ROOT_PATH));
```

Comment:
Do you mean in the API? Because this is the purpose of this method - we assume default path and only get the only parameter that is not optional (key). Can you be more specific, sorry about that

Reply:
like this;

```
    @Override
    public RedisFuture<List<Long>> jsonArrlen(K key) {
        return jsonArrlen(key, JsonPath.ROOT_PATH));
    }
```

Reply:
Let's leave it like this for now.
There is no major benefit I see in the change, but I will have it in mind if we start adding more commands (e.g. for search). And we can refactor this relatively easily (not part of the API)